test/cpp/test_tensor.cpp

```diff
@@ -370,6 +370,54 @@ TEST_F(TensorTest, TestAvgPool2DNonSquare) {
   }
 }
 
+TEST_F(TensorTest, TestBatchNorm1D) {
```

Comment:
I made a comment in #913 already.
Do we really need this?
Why would a backward plus gradient computation matter in eval mode?
Why would we also have a test in test_tensor.cpp for this (instead of test_aten_xla_tensor.cpp)?

Reply:
Hi @dlibenzi , `.eval()` is used to switch between BN layer behavior. Although rare, people sometimes use it in training mode. e.g. https://stackoverflow.com/questions/57323023/pytorch-loss-backward-and-optimizer-step-in-eval-mode-with-batch-norm-laye?rq=1

Another reason I want to fix this is - it broke our assumption that commenting out xla override and code should still work. In this case, I commented out `native_batch_norm_backward` and it was broken because our forward returns non-null `save_mean` but pytorch was expecting an undefined tensor. 

I put it in `test_tensor.cpp` mainly because `at::native_batch_norm` returns a tuple of 3 tensors and the last 2 are the ones I want to check. `torch::batch_norm` tested in `test_aten_xla_tensor.cpp` only returns 1 `torch::Tensor`.

Reply:
> Hi @dlibenzi , `.eval()` is used to switch between BN layer behavior. Although rare, people sometimes use it in backward. e.g. https://stackoverflow.com/questions/57323023/pytorch-loss-backward-and-optimizer-step-in-eval-mode-with-batch-norm-laye?rq=1

This is easy, and short fix. But one post on StackOverflow should not be reason to change code otherwise ;)


> Another reason I want to fix this is - it broke our assumption that commenting out xla override and code should still work. In this case, I commented out `native_batch_norm_backward` and it was broken because our forward returns non-null `save_mean` but pytorch was expecting an undefined tensor.

Hmm, if we override FWD and BKW, commenting one override and breaking it should not be cause of changing code.